test/integration/node-specific/bson-options/ignore_undefined.test.ts

```diff
@@ -166,206 +121,132 @@ describe('Ignore Undefined', function () {
       ignoreUndefined: false
     });
 
-    collection.insert({ a: 1, b: 2 }, err => {
-      expect(err).to.not.exist;
-
-      // Replace the doument, ignoring undefined fields
-      collection.findOneAndReplace({}, { a: 1, b: undefined }, { ignoreUndefined: true }, err => {
-        expect(err).to.not.exist;
-
-        // Locate the doument
-        collection.findOne((err, item) => {
-          expect(err).to.not.exist;
-          expect(item).to.have.property('a', 1);
-          expect(item).to.not.have.property('b');
-          done();
-        });
-      });
-    });
+    await collection.insertOne({ a: 1, b: 2 });
+
+    // Replace the doument, ignoring undefined fields
+    await collection.findOneAndReplace({}, { a: 1, b: undefined }, { ignoreUndefined: true });
+
+    const item = await collection.findOne();
+    expect(item).to.have.property('a', 1);
+    expect(item).to.not.have.property('b');
   });
 
-  it('Should correctly ignore undefined field during bulk write', function (done) {
+  it('Should correctly ignore undefined field during bulk write', async function () {
     const db = client.db('shouldCorrectlyIgnoreUndefinedValue7');
     const collection = db.collection('shouldCorrectlyIgnoreUndefinedValue7');
 
     // Ignore the undefined field
-    collection.bulkWrite(
-      [{ insertOne: { a: 1, b: undefined } }],
-      { ignoreUndefined: true },
-      err => {
-        expect(err).to.not.exist;
-
-        // Locate the doument
-        collection.findOne((err, item) => {
-          expect(err).to.not.exist;
-          expect(item).to.have.property('a', 1);
-          expect(item).to.not.have.property('b');
-          done();
-        });
-      }
-    );
+    await collection.bulkWrite([{ insertOne: { document: { a: 1, b: undefined } } }], {
+      ignoreUndefined: true
+    });
+
+    const item = await collection.findOne();
+    expect(item).to.have.property('a', 1);
+    expect(item).to.not.have.property('b');
   });
 
   describe('ignoreUndefined A server', function () {
-    it('should correctly execute insert culling undefined', {
-      metadata: { requires: { mongodb: '>=3.2' } },
-      test: function (done) {
-        const coll = client.db().collection('insert1');
-        coll.drop(() => {
-          const objectId = new ObjectId();
-          coll.insertOne(
-            { _id: objectId, a: 1, b: undefined },
-            { ignoreUndefined: true },
-            (err, res) => {
-              expect(err).to.not.exist;
-              expect(res).property('insertedId').to.exist;
-
-              const cursor = coll.find({ _id: objectId });
-              this.defer(() => cursor.close());
-
-              cursor.next((err, doc) => {
-                expect(err).to.not.exist;
-                expect(doc).to.not.have.property('b');
-                done();
-              });
-            }
-          );
-        });
-      }
+    it('should correctly execute insert culling undefined', async function () {
```

Comment:
Aren't these two tests essentially duplicates of the first test in this file?  Suggest removing them.

Reply:
The only difference I spot here is that in the first test we use `ignoreUndefined` on client level, but here on the command level, and the client doesn't have this option at all. Is it worth to test separately?
```
await coll.insertOne(
  { _id: objectId, a: 1, b: undefined },
  { ignoreUndefined: true }
);
```

Reply:
`ignoreUndefined` doesn't seem to be a TS option but it is valid, and would work if specified in the URI as well.  So probably worth keeping both tests